main/dns-root-hints/APKBUILD

```diff
@@ -13,27 +13,34 @@ options="net"
 source="verisign-grs-nstld-key.asc named.root named.root.sig"
 builddir="$srcdir/$pkgname-$pkgver"
 
-build() {
-	mkdir -p "$builddir"
+_check_sig() {
+	local GNUPGHOME="$builddir/.gpg"
+	install -d -m 0700 "$GNUPGHOME"
+	gpg --import < "$srcdir"/verisign-grs-nstld-key.asc
+	gpg --verify "${_tmp}"/named.root.sig "${_tmp}"/named.root
+	rm -rf "$GNUPGHOME"
+	mv "${_tmp}"/named.root.sig "${_tmp}"/named.root "$builddir"/
 }
 
-package() {
+_fetch_new_version() {
+	local _tmp="$(mktemp -d -p .)"
+	cd "${_tmp}"
+	curl -sSRLO "$url/named.root{.sig,}"
+	cd -
+	_check_sig
+	rm -rf "${_tmp}"
+}
+
+build() {
+	mkdir -p "$builddir"
 	cd "$builddir"
-	install -D -m 644 -o root -g root "$srcdir"/named.root \
-		"$pkgdir"/usr/share/$pkgname/named.root
-	install -D -m 644 -o root -g root "$srcdir"/named.root.sig \
-		"$pkgdir"/usr/share/$pkgname/named.root.sig
-	install -D -m 644 -o root -g root "$srcdir"/verisign-grs-nstld-key.asc \
-		"$pkgdir"/usr/share/$pkgname/verisign-grs-nstld-key.asc
 
-	# compatibility links
-	cd "$pkgdir/usr/share/$pkgname"
-	ln -s named.root named.cache
-	ln -s named.root db.cache
+	_fetch_new_version
+
+	sed -e 's/[[:space:]]\+$//' named.root > root-ns.hints
```

Comment:
We prefer to use the original zone file, without touching it. It this way the user can verify anytime the signature of the zone file used in production.

Reply:
That preference does not make a lot of sense. We don't store other sources in the repository or install signature / hash files into the running system.

I see the verification of new versions of this file as the job of this build process, not something for the users to do.